trade-finance-manager-ui/server/controllers/utilisation-reports/add-payment/add-payment-form-values-validator.ts

```diff
@@ -0,0 +1,159 @@
+import { CURRENCY, isNonEmptyString } from '@ukef/dtfs2-common';
+import { isBefore, isValid, parseISO, startOfDay } from 'date-fns';
+import { REGEX } from '../../../constants';
+import { AddPaymentErrorsViewModel, AddPaymentPaymentDateErrorViewModel, ErrorSummaryViewModel } from '../../../types/view-models';
+import { AddPaymentFormValues } from '../../../types/add-payment-form-values';
+
+const isPaymentCurrencyValid = (paymentCurrency: string | undefined): boolean => {
+  return isNonEmptyString(paymentCurrency) && (Object.values(CURRENCY) as string[]).includes(paymentCurrency);
+};
+
+const isPaymentAmountValid = (paymentAmount: string | undefined): boolean => {
+  return isNonEmptyString(paymentAmount) && REGEX.CURRENCY_NUMBER_WITH_OPTIONAL_THOUSANDS_SEPARATOR_REGEX.test(paymentAmount);
+};
+
+const isAddAnotherPaymentChoiceValid = (addAnotherPaymentChoice: string | undefined): boolean => {
+  return isNonEmptyString(addAnotherPaymentChoice) && (addAnotherPaymentChoice === 'true' || addAnotherPaymentChoice === 'false');
+};
+
+const isPaymentReferenceOverFiftyCharacters = (paymentReference: string | undefined): boolean => {
+  return isNonEmptyString(paymentReference) && paymentReference.length > 50;
+};
+
+const isValidDay = (day: string) => {
+  return REGEX.INT.test(day) && Number(day) >= 1 && Number(day) <= 31;
+};
+
+const isValidMonth = (month: string) => {
+  return REGEX.INT.test(month) && Number(month) >= 1 && Number(month) <= 12;
+};
+
+const isValidYear = (year: string) => {
+  return REGEX.YEAR.test(year);
+};
```

Comment:
We shouldn't put anything in libs/common until used in more than one service to keep it from having lots of unnecessary functions, but should be pretty trivial to move in the future.

Reply:
That's fine, but in that case I think these belong in the `tfm-ui` date helpers rather than being specific to this form. If they're not put in the service-specific date helpers then I doubt people would realise that we already have this functionality and we'd end up with it being repeated quite a lot before we realise

Reply:
I think if we were going to commonise anything here, I think it should be even further out with regards to the date field as a whole. We already have some commonised date validators, mostly in portal but they don't quite do it in the correct way, and each date field does want some slightly different validation happening so I wasn't convinced on where we should draw the line and didn't want to put time into drawing it wrong...

Reply:
Yeh, that's fair. Probably best to leave this as is and just keep it in mind for future validation